### PR TITLE
Rename ``arguments`` to ``argv`` in entry point functions

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

See @cdce8p's review here: https://github.com/PyCQA/pylint/pull/5613#discussion_r776900943

### CODE REVIEW

File: pylint/__init__.py [hunk 1]
```diff
@@ -11,22 +11,22 @@
...
 from pylint.__pkginfo__ import __version__
 
 # pylint: disable=import-outside-toplevel
 
 
-def run_pylint(*, arguments: Optional[List[str]] = None):
+def run_pylint(argv: Optional[Sequence[str]] = None):
     """Run pylint
 
     Arguments can be a list of strings normally supplied as arguments on the command line
```
Comment (DanielNoord) [inline pylint/__init__.py:24]:
```suggestion
    argv can be a list of strings normally supplied as arguments on the command line
```

File: pylint/__init__.py [hunk 2]
```diff
@@ -37,24 +37,24 @@ def run_epylint():
     EpylintRun()
 
 
-def run_pyreverse(*, arguments: Optional[List[str]] = None):
+def run_pyreverse(argv: Optional[Sequence[str]] = None):
     """Run pyreverse
 
     Arguments can be a list of strings normally supplied as arguments on the command line
```
Comment (DanielNoord) [inline pylint/__init__.py:43]:
```suggestion
    argv can be a list of strings normally supplied as arguments on the command line
```

File: pylint/__init__.py [hunk 3]
```diff
@@ -37,24 +37,24 @@ def run_epylint():
...
 
-    PyreverseRun(arguments or sys.argv[1:])
+    PyreverseRun(argv or sys.argv[1:])
 
 
-def run_symilar(*, arguments: Optional[List[str]] = None):
+def run_symilar(argv: Optional[Sequence[str]] = None):
     """Run symilar
 
     Arguments can be a list of strings normally supplied as arguments on the command line
```
Comment (DanielNoord) [inline pylint/__init__.py:53]:
```suggestion
    argv can be a list of strings normally supplied as arguments on the command line
```

File: pylint/__init__.py [hunk 4]
```diff
@@ -11,22 +11,22 @@
...
 
 # pylint: disable=import-outside-toplevel
 
 
-def run_pylint(*, arguments: Optional[List[str]] = None):
+def run_pylint(argv: Optional[Sequence[str]] = None):
     """Run pylint
 
-    Arguments can be a list of strings normally supplied as arguments on the command line
+    argv can be a list of strings normally supplied as arguments on the command line
```
Comment (cdce8p) [inline pylint/__init__.py:24]:
```suggestion
    argv can be a sequence of strings normally supplied as arguments on the command line
```
Hmm 🤔 It doesn't need to be a list, although it obviously can. E.g. a tuple would also be fine.

File: tests/test_pylint_runners.py [hunk 5]
```diff
@@ -30,5 +30,5 @@ def test_runner_with_arguments(runner: Callable, tmpdir: LocalPath) -> None:
     testargs = [filepath]
     with tmpdir.as_cwd():
         with pytest.raises(SystemExit) as err:
-            runner(arguments=testargs)
+            runner(argv=testargs)
```
Comment (cdce8p) [inline tests/test_pylint_runners.py:33]:
```suggestion
            runner(testargs)
```

**Review by cdce8p — APPROVED** [review]
LGTM! What do you think @Pierre-Sassoulas?

File: pylint/__init__.py [hunk 6]
```diff
@@ -37,24 +37,24 @@ def run_epylint():
     EpylintRun()
 
 
-def run_pyreverse(*, arguments: Optional[List[str]] = None):
+def run_pyreverse(argv: Optional[Sequence[str]] = None):
     """Run pyreverse
 
-    Arguments can be a list of strings normally supplied as arguments on the command line
+    argv can be a list of strings normally supplied as arguments on the command line
```
Comment (DanielNoord) [inline pylint/__init__.py:43]:
```suggestion
    argv can be a sequence of strings normally supplied as arguments on the command line
```

File: pylint/__init__.py [hunk 7]
```diff
@@ -37,24 +37,24 @@ def run_epylint():
...
-    PyreverseRun(arguments or sys.argv[1:])
+    PyreverseRun(argv or sys.argv[1:])
 
 
-def run_symilar(*, arguments: Optional[List[str]] = None):
+def run_symilar(argv: Optional[Sequence[str]] = None):
     """Run symilar
 
-    Arguments can be a list of strings normally supplied as arguments on the command line
+    argv can be a list of strings normally supplied as arguments on the command line
```
Comment (DanielNoord) [inline pylint/__init__.py:53]:
```suggestion
    argv can be a sequence of strings normally supplied as arguments on the command line
```

**Review by Pierre-Sassoulas — APPROVED** [review]
I think you caught something important that I missed in the last review, good job :)

**Pull request merged**